scopesim/effects/effects.py

```diff
@@ -105,9 +108,193 @@ def include(self):
     def include(self, item):
         self.meta["include"] = item
 
-    def __repr__(self):
-        name = self.meta["name"]
-        if name == "<empty>":
-            name = self.meta["filename"]
+    @property
+    def meta_string(self):
+        meta_str = ""
+        max_key_len = max([len(key) for key in self.meta.keys()])
+        for key in self.meta:
+            if key not in ["comments", "changes", "description", "history",
+                           "report_table_caption", "report_plot_caption"]:
+                meta_str += "    {} : {}\n".format(key.rjust(max_key_len),
+                                                   self.meta[key])
+    
+        return meta_str
+
+    def report(self, filename=None, output="rst", rst_title_chars="*+",
```

Comment:
This is so much cleaner than my rst and figures code, well done

Reply:
Thanks ;)